### PR TITLE
SAN-3826; Better Handling of Special Characters in Script "Replace"

### PR DESCRIPTION
**Pull request by rsandor**
Modifies the output script to handle the following:
- Allows for single quotes
- Better handling of backslashes
- Better execution of grep (without eval)

Also factored out the script tests so it is easier to test specific functional cases.
### Reviewers
- [x] @Myztiq 
- [x] @podviaznikov

### CODE REVIEW

File: lib/fs-driver.js [hunk 1]
```diff
@@ -30,7 +30,9 @@ class FsDriver {
     if (!isString(str)) {
       str = str.toString()
     }
-    return str.replace(/(['/\\])/g, '\\$1')
+    return str.replace(/'/g, '\'"\'"\'')
```
Comment (Myztiq) [inline lib/fs-driver.js]: replace all `'` with ` ' " ' " ' `?

Comment (rsandor) [inline lib/fs-driver.js]: Unfortunately [yes](http://stackoverflow.com/questions/1250079/how-to-escape-single-quotes-within-single-quoted-strings)

Comment (rsandor) [inline lib/fs-driver.js]: So roughly what this is doing is using juxtaposition in the command to "emulate" quote escaping. The first `'` ends the single quote, the next sequence is `"'"` which wraps double quotes around a single quote so it is output as is, then then final `'` opens the single-quote again.

Comment (Myztiq) [inline lib/fs-driver.js]: I don't understand why all the quotes and why we can't use `\'` or something?

Comment (podviaznikov) [inline lib/fs-driver.js]: anyway, @rsandor can you put some comment what you are doing here and why. Would be helpful for the future

Comment (Myztiq) [inline lib/fs-driver.js]: Can you put a link to that SO question, or some basic explanation above it in comments

**Pull request merged**